### PR TITLE
Add info about branding to the readme file

### PR DESCRIPTION
**Pull request by RomanNikitenko**
Add info about branding to the `readme` file

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,6 +85,59 @@ git fetch upstream-code main
...
+- `codiconCssFilePath` - the codicon css file path, should contain css styles for `codicon`s, content of the file is appended to [the coressponding css file](https://github.com/che-incubator/che-code/blob/main/code/src/vs/base/browser/ui/codicons/codicon/codicon.css), please see an example of the content [here](https://github.com/che-incubator/che-code/blob/803b864e8411bd57d617dabddfd8a132fac6c743/code/src/vs/base/browser/ui/codicons/codicon/codicon.css#L29-L33)
+
+Please note:
+-  it's expected that all paths in the `branding/product.json` file are relative to the `branding` folder. 
+For example, `workbenchConfigFilePath` field has `anyFolder/myConfigFiles/workbench-config.json` value. 
+It means that the config file can be found by the path: `che-code/branding/anyFolder/myConfigFiles/workbench-config.json`
+- At the moment the [branding.sh script](https://github.com/che-incubator/che-code/blob/main/branding/branding.sh) is not run automatically (at building project, for example), it should be integrated in the project workflow.
```
Comment (azatsarynnyy) [inline README.md]: > integrated in the project workflow

What does the project workflow mean? Don't you mind to restate it a bit?

Comment (RomanNikitenko) [reply README.md]: Replacing 
> it should be integrated in the project workflow

by 

> it should be integrated into the build process of the project

is it more clear from your point of view?

Comment (azatsarynnyy) [reply README.md]: `build process` is clear :+1: 
Thanks.

Comment (nickboldt) [inline README.md]:
```suggestion
- At the moment the [branding.sh script](https://github.com/che-incubator/che-code/blob/main/branding/branding.sh) is not run automatically (at building project, for example), it should be integrated into the build process of the project.
```

File: README.md [hunk 2]
```diff
@@ -85,6 +85,59 @@ git fetch upstream-code main
 4. Fix the conflicts or other errors. **Note**, that`./rebase.sh` script also apllies the patches from the [`.rebase`](https://github.com/che-incubator/che-code/tree/main/.rebase) directory. Sometimes, it also requires some updates there.
 5. Open a PR with your changes.
 
+## Whitelabel/branding
+ 
+There is an ability to apply branding for some UI elements. The original data can be overridden by execution [branding.sh script](https://github.com/che-incubator/che-code/blob/main/branding/branding.sh). The script expects that the branding resources are placed in the [branding folder](https://github.com/che-incubator/che-code/tree/main/branding). 
+
+The crucial one is: `product.json` file.
```
Comment (nickboldt) [inline README.md]:
```suggestion
The crucial one is: `branding/product.json` file. 
```

File: README.md [hunk 3]
```diff
@@ -85,6 +85,59 @@ git fetch upstream-code main
...
+There is an ability to apply branding for some UI elements. The original data can be overridden by execution [branding.sh script](https://github.com/che-incubator/che-code/blob/main/branding/branding.sh). The script expects that the branding resources are placed in the [branding folder](https://github.com/che-incubator/che-code/tree/main/branding). 
+
+The crucial one is: `product.json` file. 
+The values defined in the `branding/product.json` file overwrite [default ones](https://github.com/che-incubator/che-code/blob/main/code/product.json). 
+Also the file serves data mapping for provided branding resources.
+
+Example of the `branding/product.json` file:
+```
+{
+	"nameShort": "Visual Studio Code - Open Source",
```
Comment (nickboldt) [inline README.md]:
```suggestion
	"nameShort": "VS Code - Open Source",
```

File: README.md [hunk 4]
```diff
@@ -85,6 +85,59 @@ git fetch upstream-code main
...
+	"nameLong": "Red Hat OpenShift Dev Spaces - Visual Studio Code - Open Source IDE container",
```
Comment (nickboldt) [inline README.md]:
```suggestion
	"nameLong": "Red Hat OpenShift Dev Spaces with Microsoft Visual Studio Code - Open Source IDE",
```

File: README.md [hunk 5]
```diff
@@ -85,6 +85,59 @@ git fetch upstream-code main
...
+- `codiconCssFilePath` - the codicon css file path, should contain css styles for `codicon`s, content of the file is appended to [the coressponding css file](https://github.com/che-incubator/che-code/blob/main/code/src/vs/base/browser/ui/codicons/codicon/codicon.css), please see an example of the content [here](https://github.com/che-incubator/che-code/blob/803b864e8411bd57d617dabddfd8a132fac6c743/code/src/vs/base/browser/ui/codicons/codicon/codicon.css#L29-L33)
+
+Please note:
+-  it's expected that all paths in the `branding/product.json` file are relative to the `branding` folder. 
+For example, `workbenchConfigFilePath` field has `anyFolder/myConfigFiles/workbench-config.json` value. 
+It means that the config file can be found by the path: `che-code/branding/anyFolder/myConfigFiles/workbench-config.json`
+- At the moment the [branding.sh script](https://github.com/che-incubator/che-code/blob/main/branding/branding.sh) is not run automatically (at building project, for example), it should be integrated into the build process of the project.
```
Comment (nickboldt) [inline README.md]:
```suggestion
- At the moment the [branding.sh script](https://github.com/che-incubator/che-code/blob/main/branding/branding.sh) is not run automatically when building this project. It needs to be integrated into the build process of the [downstream branded project or product](https://github.com/redhat-developer/devspaces-images/blob/devspaces-3-rhel-8/devspaces-code/build/scripts/sync.sh#L96).
```

File: README.md [hunk 6]
```diff
@@ -85,6 +85,59 @@ git fetch upstream-code main
...
+- `letterpress` - icon for the main area when all editors are closed, it's possible to provide different icons for the `light` and `dark` theme  
+- `remoteIndicatorCommands` - names of commands provided by [`Eclipse Che Remote` extension](https://github.com/che-incubator/che-code/blob/main/code/extensions/che-remote/package.nls.json)
+- `workbenchConfigFilePath` - the config file path, please see an example of the config file [here](https://github.com/che-incubator/che-code/blob/main/code/src/vs/code/browser/workbench/che/workbench-config.json)
+- `codiconCssFilePath` - the codicon css file path, should contain css styles for `codicon`s, content of the file is appended to [the coressponding css file](https://github.com/che-incubator/che-code/blob/main/code/src/vs/base/browser/ui/codicons/codicon/codicon.css), please see an example of the content [here](https://github.com/che-incubator/che-code/blob/803b864e8411bd57d617dabddfd8a132fac6c743/code/src/vs/base/browser/ui/codicons/codicon/codicon.css#L29-L33)
+
+Please note:
+-  it's expected that all paths in the `branding/product.json` file are relative to the `branding` folder.
```
Comment (nickboldt) [inline README.md]:
```suggestion
-  All paths in the `branding/product.json` file must be relative to the `branding` folder. 
```

File: README.md (hunk 4 above)
Comment (nickboldt) [reply README.md]: Update per CRW-3098.

File: README.md (hunk 3 above)
Comment (nickboldt) [reply README.md]: Update per CRW-3098.

File: README.md (hunk 5 above)
Comment (max-cx) [reply README.md]:
```suggestion
- Currently, the [branding.sh](https://github.com/che-incubator/che-code/blob/main/branding/branding.sh) script is not run automatically when building this project. (In the future, this script will be integrated into the build process of the [downstream product](https://github.com/redhat-developer/devspaces-images/blob/devspaces-3-rhel-8/devspaces-code/build/scripts/sync.sh#L96).)
```

File: README.md [hunk 7]
```diff
@@ -85,6 +85,59 @@ git fetch upstream-code main
...
+- `remoteIndicatorCommands` - names of commands provided by [`Eclipse Che Remote` extension](https://github.com/che-incubator/che-code/blob/main/code/extensions/che-remote/package.nls.json)
+- `workbenchConfigFilePath` - the config file path, please see an example of the config file [here](https://github.com/che-incubator/che-code/blob/main/code/src/vs/code/browser/workbench/che/workbench-config.json)
+- `codiconCssFilePath` - the codicon css file path, should contain css styles for `codicon`s, content of the file is appended to [the coressponding css file](https://github.com/che-incubator/che-code/blob/main/code/src/vs/base/browser/ui/codicons/codicon/codicon.css), please see an example of the content [here](https://github.com/che-incubator/che-code/blob/803b864e8411bd57d617dabddfd8a132fac6c743/code/src/vs/base/browser/ui/codicons/codicon/codicon.css#L29-L33)
+
+Please note:
+-  it's expected that all paths in the `branding/product.json` file are relative to the `branding` folder. 
+For example, `workbenchConfigFilePath` field has `anyFolder/myConfigFiles/workbench-config.json` value.
```
Comment (max-cx) [inline README.md]:
```suggestion
For example, the `workbenchConfigFilePath` field might have the `anyFolder/myConfigFiles/workbench-config.json` value. 
```

File: README.md [hunk 8]
```diff
@@ -85,6 +85,59 @@ git fetch upstream-code main
...
+- `letterpress` - icon for the main area when all editors are closed, it's possible to provide different icons for the `light` and `dark` theme  
+- `remoteIndicatorCommands` - names of commands provided by [`Eclipse Che Remote` extension](https://github.com/che-incubator/che-code/blob/main/code/extensions/che-remote/package.nls.json)
+- `workbenchConfigFilePath` - the config file path, please see an example of the config file [here](https://github.com/che-incubator/che-code/blob/main/code/src/vs/code/browser/workbench/che/workbench-config.json)
+- `codiconCssFilePath` - the codicon css file path, should contain css styles for `codicon`s, content of the file is appended to [the coressponding css file](https://github.com/che-incubator/che-code/blob/main/code/src/vs/base/browser/ui/codicons/codicon/codicon.css), please see an example of the content [here](https://github.com/che-incubator/che-code/blob/803b864e8411bd57d617dabddfd8a132fac6c743/code/src/vs/base/browser/ui/codicons/codicon/codicon.css#L29-L33)
+
+Please note:
```
Comment (max-cx) [inline README.md]:
```suggestion
NOTE:
```

File: README.md [hunk 9]
```diff
@@ -85,6 +85,59 @@ git fetch upstream-code main
...
+}
+```
+- `nameShort` - application name
+- `nameLong` - it's used for the `Welcome` page, the `About` dialog, browser tab title
+- `favicon` - icon for the browser tab title, the same for all themes
+- `welcome` - icon for the `Welcome` (`Get Started`) page tab title, the same for all themes
+- `statusBarItem` - icon for the status bar item, the same for all themes, should be defined as `codicon` in the `workbench-config.json` file and `codicon` css styles as well
+- `letterpress` - icon for the main area when all editors are closed, it's possible to provide different icons for the `light` and `dark` theme  
+- `remoteIndicatorCommands` - names of commands provided by [`Eclipse Che Remote` extension](https://github.com/che-incubator/che-code/blob/main/code/extensions/che-remote/package.nls.json)
+- `workbenchConfigFilePath` - the config file path, please see an example of the config file [here](https://github.com/che-incubator/che-code/blob/main/code/src/vs/code/browser/workbench/che/workbench-config.json)
```
Comment (max-cx) [inline README.md]:
```suggestion
- `workbenchConfigFilePath` - The config file path. See an [example of the config file](https://github.com/che-incubator/che-code/blob/main/code/src/vs/code/browser/workbench/che/workbench-config.json).
```

File: README.md [hunk 10]
```diff
@@ -85,6 +85,59 @@ git fetch upstream-code main
...
+	"codiconCssFilePath": "css/codicon.css"
+}
+```
+- `nameShort` - application name
+- `nameLong` - it's used for the `Welcome` page, the `About` dialog, browser tab title
+- `favicon` - icon for the browser tab title, the same for all themes
+- `welcome` - icon for the `Welcome` (`Get Started`) page tab title, the same for all themes
+- `statusBarItem` - icon for the status bar item, the same for all themes, should be defined as `codicon` in the `workbench-config.json` file and `codicon` css styles as well
+- `letterpress` - icon for the main area when all editors are closed, it's possible to provide different icons for the `light` and `dark` theme  
+- `remoteIndicatorCommands` - names of commands provided by [`Eclipse Che Remote` extension](https://github.com/che-incubator/che-code/blob/main/code/extensions/che-remote/package.nls.json)
```
Comment (max-cx) [inline README.md]:
```suggestion
- `remoteIndicatorCommands` - The names of commands provided by the [`Eclipse Che Remote`](https://github.com/che-incubator/che-code/blob/main/code/extensions/che-remote/package.nls.json) extension.
```

File: README.md [hunk 11]
```diff
@@ -85,6 +85,59 @@ git fetch upstream-code main
...
+	"workbenchConfigFilePath": "workbench-config.json",
+	"codiconCssFilePath": "css/codicon.css"
+}
+```
+- `nameShort` - application name
+- `nameLong` - it's used for the `Welcome` page, the `About` dialog, browser tab title
+- `favicon` - icon for the browser tab title, the same for all themes
+- `welcome` - icon for the `Welcome` (`Get Started`) page tab title, the same for all themes
+- `statusBarItem` - icon for the status bar item, the same for all themes, should be defined as `codicon` in the `workbench-config.json` file and `codicon` css styles as well
+- `letterpress` - icon for the main area when all editors are closed, it's possible to provide different icons for the `light` and `dark` theme
```
Comment (max-cx) [inline README.md]:
```suggestion
- `letterpress` - The icon for the main area when all editors are closed. It's possible to provide different icons for `light` and `dark` themes.
```

File: README.md [hunk 12]
```diff
@@ -85,6 +85,59 @@ git fetch upstream-code main
...
+	},
+	"workbenchConfigFilePath": "workbench-config.json",
+	"codiconCssFilePath": "css/codicon.css"
+}
+```
+- `nameShort` - application name
+- `nameLong` - it's used for the `Welcome` page, the `About` dialog, browser tab title
+- `favicon` - icon for the browser tab title, the same for all themes
+- `welcome` - icon for the `Welcome` (`Get Started`) page tab title, the same for all themes
+- `statusBarItem` - icon for the status bar item, the same for all themes, should be defined as `codicon` in the `workbench-config.json` file and `codicon` css styles as well
```
Comment (max-cx) [inline README.md]:
```suggestion
- `statusBarItem` - The icon for the status bar item. It's the same for all themes and must be defined as `codicon` in the `workbench-config.json` file and the `codicon` CSS styles.
```

File: README.md [hunk 13]
```diff
@@ -85,6 +85,59 @@ git fetch upstream-code main
...
+- `letterpress` - icon for the main area when all editors are closed, it's possible to provide different icons for the `light` and `dark` theme  
+- `remoteIndicatorCommands` - names of commands provided by [`Eclipse Che Remote` extension](https://github.com/che-incubator/che-code/blob/main/code/extensions/che-remote/package.nls.json)
+- `workbenchConfigFilePath` - the config file path, please see an example of the config file [here](https://github.com/che-incubator/che-code/blob/main/code/src/vs/code/browser/workbench/che/workbench-config.json)
+- `codiconCssFilePath` - the codicon css file path, should contain css styles for `codicon`s, content of the file is appended to [the coressponding css file](https://github.com/che-incubator/che-code/blob/main/code/src/vs/base/browser/ui/codicons/codicon/codicon.css), please see an example of the content [here](https://github.com/che-incubator/che-code/blob/803b864e8411bd57d617dabddfd8a132fac6c743/code/src/vs/base/browser/ui/codicons/codicon/codicon.css#L29-L33)
```
Comment (max-cx) [inline README.md]:
```suggestion
- `codiconCssFilePath` - The codicon css file path. Must contain CSS styles for `codicon`s. The content of the file is appended to [the coressponding css file](https://github.com/che-incubator/che-code/blob/main/code/src/vs/base/browser/ui/codicons/codicon/codicon.css). See an [example of the content](https://github.com/che-incubator/che-code/blob/803b864e8411bd57d617dabddfd8a132fac6c743/code/src/vs/base/browser/ui/codicons/codicon/codicon.css#L29-L33).
```

File: README.md [hunk 14]
```diff
@@ -85,6 +85,59 @@ git fetch upstream-code main
...
+
+Please note:
+-  it's expected that all paths in the `branding/product.json` file are relative to the `branding` folder. 
+For example, `workbenchConfigFilePath` field has `anyFolder/myConfigFiles/workbench-config.json` value. 
+It means that the config file can be found by the path: `che-code/branding/anyFolder/myConfigFiles/workbench-config.json`
```
Comment (max-cx) [inline README.md]:
```suggestion
This means that the config file can be found by the path: `che-code/branding/anyFolder/myConfigFiles/workbench-config.json`
```

File: README.md [hunk 15]
```diff
@@ -85,6 +85,59 @@ git fetch upstream-code main
...
+		"stopWorkspaceCommand": "Dev Spaces: Stop Workspace"
+	},
+	"workbenchConfigFilePath": "workbench-config.json",
+	"codiconCssFilePath": "css/codicon.css"
+}
+```
+- `nameShort` - application name
+- `nameLong` - it's used for the `Welcome` page, the `About` dialog, browser tab title
+- `favicon` - icon for the browser tab title, the same for all themes
+- `welcome` - icon for the `Welcome` (`Get Started`) page tab title, the same for all themes
```
Comment (max-cx) [inline README.md]:
```suggestion
- `welcome` - The icon for the **Welcome** (**Get Started**) page tab title. It's the same for all themes.
```

File: README.md [hunk 16]
```diff
@@ -85,6 +85,59 @@ git fetch upstream-code main
...
+		"openDashboardCommand": "Dev Spaces: Open Dashboard",
+		"stopWorkspaceCommand": "Dev Spaces: Stop Workspace"
+	},
+	"workbenchConfigFilePath": "workbench-config.json",
+	"codiconCssFilePath": "css/codicon.css"
+}
+```
+- `nameShort` - application name
+- `nameLong` - it's used for the `Welcome` page, the `About` dialog, browser tab title
+- `favicon` - icon for the browser tab title, the same for all themes
```
Comment (max-cx) [inline README.md]:
```suggestion
- `favicon` - The icon for the browser tab title. It's the same for all themes.
```

File: README.md [hunk 17]
```diff
@@ -85,6 +85,59 @@ git fetch upstream-code main
...
+		"openDocumentationCommand": "Dev Spaces: Open Documentation",
+		"openDashboardCommand": "Dev Spaces: Open Dashboard",
+		"stopWorkspaceCommand": "Dev Spaces: Stop Workspace"
+	},
+	"workbenchConfigFilePath": "workbench-config.json",
+	"codiconCssFilePath": "css/codicon.css"
+}
+```
+- `nameShort` - application name
+- `nameLong` - it's used for the `Welcome` page, the `About` dialog, browser tab title
```
Comment (max-cx) [inline README.md]:
```suggestion
- `nameLong` - This is used for the **Welcome** page, the **About** dialog, and browser tab title.
```

File: README.md [hunk 18]
```diff
@@ -85,6 +85,59 @@ git fetch upstream-code main
...
+	"remoteIndicatorCommands": {
+		"openDocumentationCommand": "Dev Spaces: Open Documentation",
+		"openDashboardCommand": "Dev Spaces: Open Dashboard",
+		"stopWorkspaceCommand": "Dev Spaces: Stop Workspace"
+	},
+	"workbenchConfigFilePath": "workbench-config.json",
+	"codiconCssFilePath": "css/codicon.css"
+}
+```
+- `nameShort` - application name
```
Comment (max-cx) [inline README.md]:
```suggestion
- `nameShort` - The application name.
```

File: README.md [hunk 19]
```diff
@@ -85,6 +85,59 @@ git fetch upstream-code main
 4. Fix the conflicts or other errors. **Note**, that`./rebase.sh` script also apllies the patches from the [`.rebase`](https://github.com/che-incubator/che-code/tree/main/.rebase) directory. Sometimes, it also requires some updates there.
 5. Open a PR with your changes.
 
+## Whitelabel/branding
+ 
+There is an ability to apply branding for some UI elements. The original data can be overridden by execution [branding.sh script](https://github.com/che-incubator/che-code/blob/main/branding/branding.sh). The script expects that the branding resources are placed in the [branding folder](https://github.com/che-incubator/che-code/tree/main/branding). 
+
+The crucial one is: `product.json` file. 
+The values defined in the `branding/product.json` file overwrite [default ones](https://github.com/che-incubator/che-code/blob/main/code/product.json).
```
Comment (max-cx) [inline README.md]:
```suggestion
The values defined in the `branding/product.json` file override the [default values](https://github.com/che-incubator/che-code/blob/main/code/product.json). 
```

File: README.md (hunk 2 above)
Comment (max-cx) [inline README.md]:
```suggestion
The `branding/product.json` file is crucial.
```

File: README.md [hunk 20]
```diff
@@ -85,6 +85,59 @@ git fetch upstream-code main
 4. Fix the conflicts or other errors. **Note**, that`./rebase.sh` script also apllies the patches from the [`.rebase`](https://github.com/che-incubator/che-code/tree/main/.rebase) directory. Sometimes, it also requires some updates there.
 5. Open a PR with your changes.
 
+## Whitelabel/branding
+ 
+There is an ability to apply branding for some UI elements. The original data can be overridden by execution [branding.sh script](https://github.com/che-incubator/che-code/blob/main/branding/branding.sh). The script expects that the branding resources are placed in the [branding folder](https://github.com/che-incubator/che-code/tree/main/branding).
```
Comment (max-cx) [inline README.md]:
```suggestion
There is a functionality to apply branding to some UI elements. The original data can be overridden by executing the [branding.sh](https://github.com/che-incubator/che-code/blob/main/branding/branding.sh) script. The script searches for the branding resources in the [branding folder](https://github.com/che-incubator/che-code/tree/main/branding). 
```

File: README.md (hunk 5 above)
Comment (RomanNikitenko) [reply README.md]: > In the future, this script will be integrated

If we speak about `devspaces-code` - it's already integrated, please see [here](https://github.com/redhat-developer/devspaces-images/blob/devspaces-3-rhel-8/devspaces-code/build/scripts/sync.sh#L96)

In my sentence above I meant that the script should be integrated for other projects based on the `che-code`

File: README.md (hunk 2 above)
Comment (RomanNikitenko) [reply README.md]: thank you, applied manually as there was a conflict

**Comment by RomanNikitenko** [comment]
@nickboldt @max-cx 
thank you very much for the review, I applied all your suggested changes!

**Review by azatsarynnyy — APPROVED** [review]

**Review by nickboldt — APPROVED** [review]

**Pull request merged**